tools/c7n_azure/tests_azure/test_resource_graph_source.py

```diff
@@ -0,0 +1,141 @@
+            'resource': 'azure.storage',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctstorage*'}]
+        })
+
+        resources_arm = p1.run()[0]
+
+        p2 = self.load_policy({
+            'name': 'test-azure-storage-arm-source',
+            'resource': 'azure.storage',
+            'source': 'resource-graph',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctstorage*'}]
+        }, validate=True)
+
+        resources_resource_graph = p2.run()[0]
+
+        self.assertTrue(resource_cmp(resources_arm, resources_resource_graph))
+
+    @arm_template('vm.json')
+    def test_resource_graph_and_arm_sources_vm_are_equivalent(self):
+        p1 = self.load_policy({
+            'name': 'test-azure-vm-arm-source',
+            'resource': 'azure.vm',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctestvm*'}]
+        })
+
+        resources_arm = p1.run()[0]
+
+        p2 = self.load_policy({
+            'name': 'test-azure-vm-arm-source',
+            'resource': 'azure.vm',
+            'source': 'resource-graph',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'cctestvm*'}]
+        }, validate=True)
+
+        resources_resource_graph = p2.run()[0]
+
+        # ARM returns the vm child extension resources that is not return with the resource graph
+        self.assertTrue(
+            resource_cmp(resources_arm, resources_resource_graph, ignore_properties=['resources']))
+
+
+def resource_cmp(res1, res2, ignore_properties=[]):
+    """
+    Recursively compare for equality the resources return by ARM to the Resource Graph.
+    Resource Graph has more properties than the default ARM call.
+    :param ignore_properties: list of properties we can skip comparing
+    :param res1: dictionary that represents the ARM resource
+    :param res2: dictionary that represents the Resource Graph resource
+    :return: True if every property for ARM is returned in the Resource Graph, else False
+    """
+    if type(res1) != type(res2):
+        return False
+
+    if type(res1) == dict:
+        for prop in res1:
+            if prop not in ignore_properties and \
+                    (prop not in res2 or not resource_cmp(res1[prop], res2[prop])):
+                return False
+
+    elif type(res1) == list:
+        if len(res1) != len(res2):
+            return False
+
+        for item1, item2 in zip(res1, res2):
```

Comment:
Agreed, this gets a little tricky because sometimes the element of the array is another dictionary and it's not possible to know the key ahead of time.